src/codeu/chat/common/ConversationHeader.java

```diff
@@ -44,23 +46,88 @@ public ConversationHeader read(InputStream in) throws IOException {
           Uuid.SERIALIZER.read(in),
           Uuid.SERIALIZER.read(in),
           Time.SERIALIZER.read(in),
-          Serializers.STRING.read(in)
+          Serializers.STRING.read(in),
+          Serializers.INTEGER.read(in)
       );
 
     }
   };
 
+  // permission levels
+  // member == 1
+  // owner == 2
+  // creator == 3
+  
+  public enum permissions {
```

Comment:
This is a nice enum but it doesn't really get used (except very briefly in the ConversationHeader constructor where you access creator.getPermissionLevel() (aka, "3"). There's definitely potential cleanliness to using an enum, but given that almost everywhere else you just work with the raw integer value, I would suggest in the interest of time / simplicity that you just stick with the integer representation.